### PR TITLE
Remove 3.6 support

### LINKED ISSUE #361: Remove support for python 3.6
Python 3.6 is EOL and support has been removed from https://github.com/awslabs/aws-config-rdklib/ already.

### PR DESCRIPTION
**Pull request by mbeacom**
*Issue #, if available:* #361

*Description of changes:* The goal of this PR is to remove Python 3.6 support (tests, references, etc) due to EOL.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: rdk/rdk.py [hunk 1]
```diff
@@ -4115,7 +4106,7 @@ def __tag_config_rule(self, rule_name, cfn_tags, my_session):
     def __get_lambda_layers(self, my_session, args, params):
         layers = []
         if "SourceRuntime" in params:
-            if params["SourceRuntime"] in ["python3.6-lib", "python3.7-lib", "python3.8-lib"]:
+            if params["SourceRuntime"] in ["python3.7-lib", "python3.8-lib"]:
```
Comment (bmorrissirromb) [inline rdk/rdk.py:4109]:
```suggestion
            if params["SourceRuntime"] in ["python3.7-lib", "python3.8-lib", "python3.9-lib"]:
```

File: rdk/rdk.py [hunk 2]
```diff
@@ -4239,7 +4230,7 @@ def __create_new_lambda_layer_locally(self, my_session, layer_name="rdklib-layer
         lambda_client.publish_layer_version(
             LayerName=layer_name,
             Content={"S3Bucket": bucket_name, "S3Key": layer_name},
-            CompatibleRuntimes=["python3.6", "python3.7", "python3.8"],
+            CompatibleRuntimes=["python3.7", "python3.8"],
```
Comment (bmorrissirromb) [inline rdk/rdk.py:4233]:
```suggestion
            CompatibleRuntimes=["python3.7", "python3.8", "python3.9"],
```

File: rdk/rdk.py [hunk 3]
```diff
@@ -558,7 +556,7 @@ def get_deployment_parser(ForceArgument=False, Command="deploy"):
         "--generated-lambda-layer",
         required=False,
         action="store_true",
-        help="[optional] Forces rdk deploy to use the Python(3.6-lib,3.7-lib,3.8-lib,) lambda layer generated by rdk init --generate-lambda-layer",
+        help="[optional] Forces rdk deploy to use the Python(3.7-lib,3.8-lib,) lambda layer generated by rdk init --generate-lambda-layer",
```
Comment (bmorrissirromb) [inline rdk/rdk.py:559]: Not even sure why this parenthetical is here, I doubt it helps much.

```suggestion
        help="[optional] Forces rdk deploy to use the Python(3.7-lib,3.8-lib,3.9-lib) lambda layer generated by rdk init --generate-lambda-layer",
```

File: rdk/rdk.py [hunk 4]
```diff
@@ -655,7 +653,7 @@ def get_deployment_organization_parser(ForceArgument=False, Command="deploy-orga
...
```
Comment (bmorrissirromb) [inline rdk/rdk.py:656]:
```suggestion
        help="[optional] Forces rdk deploy to use the Python(3.7-lib,3.8-lib,3.9-lib) lambda layer generated by rdk init --generate-lambda-layer",
```

File: rdk/rdk.py [hunk 5]
```diff
@@ -726,7 +724,7 @@ def get_export_parser(ForceArgument=False, Command="export"):
...
```
Comment (bmorrissirromb) [inline rdk/rdk.py:727]:
```suggestion
        help="[optional] Forces rdk deploy to use the Python(3.7-lib,3.8-lib,3.9-lib) lambda layer generated by rdk init --generate-lambda-layer",
```

File: rdk/rdk.py (hunk 1 above)
Comment (mbeacom) [reply rdk/rdk.py:4109]: @bmorrissirromb I intentionally wanted to add the missing 3.9 stubs in a subsequent PR and leave this strictly to removal of 3.6.

Is that not desired?

Comment (mbeacom) [reply rdk/rdk.py:4109]: Resolved in https://github.com/mbeacom/aws-config-rdk/compare/remove-python36-361...mbeacom:aws-config-rdk:add-python39-362?expand=1

File: rdk/rdk.py (hunk 3 above)
Comment (mbeacom) [reply rdk/rdk.py:559]: Resolved in https://github.com/mbeacom/aws-config-rdk/compare/remove-python36-361...mbeacom:aws-config-rdk:remove-python-copy-364?expand=1 -- will rebase once merged and raise PR.

File: rdk/rdk.py (hunk 5 above)
Comment (mbeacom) [reply rdk/rdk.py:727]: Resolved in https://github.com/mbeacom/aws-config-rdk/compare/remove-python36-361...mbeacom:aws-config-rdk:remove-python-copy-364?expand=1 -- will rebase once merged and raise PR.

File: rdk/rdk.py (hunk 1 above)
Comment (mbeacom) [reply rdk/rdk.py:4109]: Resolved in https://github.com/mbeacom/aws-config-rdk/compare/remove-python36-361...mbeacom:aws-config-rdk:remove-python-copy-364?expand=1 -- will rebase once merged and raise PR.

File: rdk/rdk.py (hunk 4 above)
Comment (mbeacom) [reply rdk/rdk.py:656]: Resolved in https://github.com/mbeacom/aws-config-rdk/compare/remove-python36-361...mbeacom:aws-config-rdk:remove-python-copy-364?expand=1 -- will rebase once merged and raise PR.

**Review by bmorrissirromb — APPROVED** [review]

**Pull request merged**